src/mono/mono/mini/interp/interp.c

```diff
@@ -6425,6 +6425,13 @@ MINT_IN_CASE(MINT_BRTRUE_I8_SP) ZEROP_SP(gint64, !=); MINT_IN_BREAK;
 			MonoVTable *vtable = (MonoVTable*)frame->imethod->data_items [ip [3]];
 			MonoClass *c = vtable->klass;
 
+			if (G_UNLIKELY (m_class_is_byreflike (c))) {
+				char *str = g_strdup_printf ("Cannot box IsByRefLike type '%s.%s'", m_class_get_name_space (c), m_class_get_name (c));
+				MonoException *ex = mono_get_exception_bad_image_format (str);
```

Comment:
CoreCLR throws InvalidProgramException for attempting to box a ByRefLike type. Mono should align itself with CoreCLR.